cmd/bucket-notification-handlers.go

```diff
@@ -87,15 +86,38 @@ func (api objectAPIHandlers) GetBucketNotificationHandler(w http.ResponseWriter,
 			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 			return
 		}
-		config = &event.Config{}
-	} else {
-		if err = xml.NewDecoder(bytes.NewReader(configData)).Decode(&config); err != nil {
+		notificationBytes, err := xml.Marshal(&event.Config{
+			XMLNS: "http://s3.amazonaws.com/doc/2006-03-01/",
+		})
+		if err != nil {
 			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 			return
 		}
+
+		writeSuccessResponseXML(w, notificationBytes)
+		return
 	}
 
-	config.SetRegion(globalServerRegion)
+	config := event.Config{}
+	if err = xml.Unmarshal(configData, &config); err != nil {
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+		return
+	}
+
+	if err = config.Validate(globalServerRegion, globalNotificationSys.targetList); err != nil {
+		arnErr, ok := err.(*event.ErrARNNotFound)
+		if !ok {
+			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+			return
+		}
+		for i, queue := range config.QueueList {
+			// Remove ARN not found queues
+			if reflect.DeepEqual(queue.ARN, arnErr.ARN) {
```

Comment:
This will remove the ARN from the config if it is not present in the globalNotificationSys.targetList? Is it ok if we remove?

Reply:
Attaching a comment regarding this conversation - https://github.com/minio/minio/pull/7673#issuecomment-494275324 Does this make sense?

Reply:
Yes the intention is to allow existing ones to be not sent to client which are stale. Since they are not useful to client.